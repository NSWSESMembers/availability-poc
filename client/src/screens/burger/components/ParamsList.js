import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { FlatList } from 'react-native';
import ParamsListItem from './ParamsListItem';


class ParamsList extends Component {
  renderItem = ({ item }) => (
    <ParamsListItem
      {...item}
    />
  )

  render() {
    const {
      deviceInfo,
      packageInfo,
      user,
      device,
      auth,
    } = this.props;

    const items = [
      {
        title: 'Device Info',
        detail: deviceInfo,
      },
      {
        title: 'Package Info',
        detail: packageInfo,
      },
      {
        title: 'Current auth state',
        detail: auth,
      },
      {
        title: 'Current device query',
        detail: device,
      },
      {
        title: 'Current user query',
        detail: user,
      },
    ];

    return (
      <FlatList
        data={items}
        keyExtractor={item => item.title}
        renderItem={this.renderItem}
      />
    );
  }
}

ParamsList.propTypes = {
  deviceInfo: PropTypes.string,
  packageInfo: PropTypes.string,
  auth: PropTypes.string,
  user: PropTypes.string,
  device: PropTypes.string,
};

export default ParamsList;
