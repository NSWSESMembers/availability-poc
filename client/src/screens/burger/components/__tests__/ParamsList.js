import React from 'react';
import renderer from 'react-test-renderer';

import ParamsList from '../ParamsList';

test('renders correctly', () => {
  const tree = renderer.create(
    <ParamsList
      deviceInfo="deviceInfo"
      packageInfo="packageInfo"
      auth="auth"
      device="device"
      user="user"
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
