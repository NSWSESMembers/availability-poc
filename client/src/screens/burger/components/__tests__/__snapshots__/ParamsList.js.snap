// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "detail": "deviceInfo",
        "title": "Device Info",
      },
      Object {
        "detail": "packageInfo",
        "title": "Package Info",
      },
      Object {
        "detail": "auth",
        "title": "Current auth state",
      },
      Object {
        "detail": "device",
        "title": "Current device query",
      },
      Object {
        "detail": "user",
        "title": "Current user query",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "elevation": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 6,
                "paddingBottom": 10,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "flex": 0,
                  "fontWeight": "bold",
                  "includeFontPadding": false,
                  "marginRight": 8,
                  "textAlignVertical": "top",
                }
              }
            >
              Device Info
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "color": "#6B6B6B",
                  "flex": 0,
                  "fontSize": 12,
                  "includeFontPadding": false,
                  "textAlignVertical": "bottom",
                }
              }
            >
              deviceInfo
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "elevation": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 6,
                "paddingBottom": 10,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "flex": 0,
                  "fontWeight": "bold",
                  "includeFontPadding": false,
                  "marginRight": 8,
                  "textAlignVertical": "top",
                }
              }
            >
              Package Info
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "color": "#6B6B6B",
                  "flex": 0,
                  "fontSize": 12,
                  "includeFontPadding": false,
                  "textAlignVertical": "bottom",
                }
              }
            >
              packageInfo
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "elevation": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 6,
                "paddingBottom": 10,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "flex": 0,
                  "fontWeight": "bold",
                  "includeFontPadding": false,
                  "marginRight": 8,
                  "textAlignVertical": "top",
                }
              }
            >
              Current auth state
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "color": "#6B6B6B",
                  "flex": 0,
                  "fontSize": 12,
                  "includeFontPadding": false,
                  "textAlignVertical": "bottom",
                }
              }
            >
              auth
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "elevation": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 6,
                "paddingBottom": 10,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "flex": 0,
                  "fontWeight": "bold",
                  "includeFontPadding": false,
                  "marginRight": 8,
                  "textAlignVertical": "top",
                }
              }
            >
              Current device query
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "color": "#6B6B6B",
                  "flex": 0,
                  "fontSize": 12,
                  "includeFontPadding": false,
                  "textAlignVertical": "bottom",
                }
              }
            >
              device
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "elevation": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 6,
                "paddingBottom": 10,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "flex": 0,
                  "fontWeight": "bold",
                  "includeFontPadding": false,
                  "marginRight": 8,
                  "textAlignVertical": "top",
                }
              }
            >
              Current user query
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Object {
                  "color": "#6B6B6B",
                  "flex": 0,
                  "fontSize": 12,
                  "includeFontPadding": false,
                  "textAlignVertical": "bottom",
                }
              }
            >
              user
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
